### PR TITLE
Remove node Link connection storage

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

**Comment by joshuef** [comment]
bors try

**Comment by joshuef** [comment]
bors try

File: sn_node/src/node/error.rs [hunk 1]
```diff
@@ -30,6 +30,10 @@ pub type Result<T, E = Error> = std::result::Result<T, E>;
 #[derive(Debug, Error)]
 #[allow(missing_docs)]
 pub enum Error {
+    /// Client message response would have redirected to ourselves _again_.
+    /// So we bail and attempt to update our knowledge instead
+    #[error("Node is missing network knowledge of sibling section. We should trigger AntiEntropy probe to update.")]
+    MissingSiblingNetworkKnowledge,
```
Comment (bochaco) [inline sn_node/src/node/error.rs:36]: I think having context info like the node's own/sibling section prefix will be useful in order to know which is the sibling section info missing in that node.

File: sn_client/src/api/mod.rs [hunk 2]
```diff
@@ -84,7 +84,7 @@ impl Client {
             self.session.endpoint.public_addr()
         );
 
-        let mut attempts = 1;
+        let mut attempts = 0;
```
Comment (bochaco) [inline sn_client/src/api/mod.rs:87]: This is wrong as per how we use it within the loop, and how we compare it against `NETWORK_PROBE_MAX_ATTEMPTS`

File: sn_client/src/api/mod.rs [hunk 3]
```diff
@@ -114,6 +114,10 @@ impl Client {
                         "Initial probe msg to network failed. Trying again (attempt #{}): {:?}",
                         attempts, result
                     );
+
+                    // do a tiny sleep to not hammer the network and allow any AE flows there
+                    // time to complete
+                    tokio::time::sleep(tokio::time::Duration::from_millis(500)).await
```
Comment (bochaco) [inline sn_client/src/api/mod.rs:120]: I'm not sure we need this, the AE would be all completed within the `send_query_without_retry` lifecycle before returning, although we may need a timeout around the call to `send_query_without_retry` as I think it's not bounded and could be stuck for long?

**Review by bochaco — CHANGES_REQUESTED** [review]

**Comment by joshuef** [comment]
bors try

**Comment by joshuef** [comment]
I'm going to close this, as I have the feeling it introduces some more issues in/around too many conections being spawned and slowing us down. (I've seen big pauses where we have eg 300 incoming connections in a oner... this appears to block things, though I'm not sure why atm).

https://github.com/maidsafe/safe_network/pull/1823 is a candidate to sort this in another way

**Pull request closed without merging**